### PR TITLE
Make --androidx flag a noop in flutter create

### PR DESCRIPTION
**Pull request by blasten**
## Description

Make androidx flag a noop in `flutter create`. This means that there won't be a way to create a non-androidx project from flutter create.

## Related Issues

https://github.com/flutter/flutter/issues/39283

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Were any of the IDEs configured to pass this flag?

**Comment by jmagman** [comment]
Needs to be updated in plugin_tools first for plugin CI https://github.com/flutter/plugin_tools/blob/3aea345df9220a42b9094e3034591c95eb3fa815/lib/src/create_all_plugins_app_command.dart#L57

**Comment by jmagman** [comment]
> Were any of the IDEs configured to pass this flag?

There's a checkmark that adds --no-androidx
https://github.com/flutter/flutter-intellij/blob/82d3aaa8279e44d01073120a68d29dbf3f34cb0f/src/io/flutter/sdk/FlutterCreateAdditionalSettings.java#L133
https://github.com/flutter/flutter-intellij/blob/024b84cb8ec2b723047c7794b3470478448e5fac/testSrc/unit/io/flutter/sdk/FlutterCreateAdditionalSettingsTest.java#L144

**Comment by blasten** [comment]
@stevemessick is it possible to remove the `--no-androidx` flag?

**Comment by stevemessick** [comment]
Could we do this in two stages? Allow the arg parser to accept the flag but ignore it for now. Once this gets into stable we can remove the flag. Maybe make hidden, too. Would that work? Otherwise we're going to have to do version checking in the IDE when we build that UI.

**Comment by blasten** [comment]
@stevemessick sgtm

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 1]
```diff
@@ -139,11 +139,11 @@ class CreateCommand extends FlutterCommand {
...
       allowed: <String>['java', 'kotlin'],
     );
+    // TODO(egarciad): Remove this flag. https://github.com/flutter/flutter/issues/52363
     argParser.addFlag(
       'androidx',
-      negatable: true,
-      defaultsTo: true,
-      help: 'Generate a project using the AndroidX support libraries',
+      hide: true,
+      help: 'Noop flag. (settings this flag to false has no effect)',
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/commands/create.dart]: Maybe:
'Deprecated. Setting this flag has no effect.'

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 2]
```diff
@@ -139,11 +139,11 @@ class CreateCommand extends FlutterCommand {
       defaultsTo: 'kotlin',
       allowed: <String>['java', 'kotlin'],
     );
+    // TODO(egarciad): Remove this flag. https://github.com/flutter/flutter/issues/52363
     argParser.addFlag(
       'androidx',
-      negatable: true,
-      defaultsTo: true,
-      help: 'Generate a project using the AndroidX support libraries',
+      hide: true,
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/commands/create.dart]: Does this still need to be negatable to support `--no-androidx` usage?

**Review by jmagman — APPROVED** [review]
LGTM!

**Pull request merged**